mlir/lib/Dialect/Math/Transforms/ExpandPatterns.cpp

```diff
@@ -441,6 +441,25 @@ static LogicalResult convertRoundEvenOp(math::RoundEvenOp op,
   return success();
 }
 
+// Convert `math.rsqrt` into `arith.divf` + `math.sqrt`
+static LogicalResult convertRsqrtOp(math::RsqrtOp op,
+                                    PatternRewriter &rewriter) {
+
+  auto operand = op.getOperand();
+  auto operandTy = operand.getType();
+  auto eTy = getElementTypeOrSelf(operandTy);
+  if (!isa<FloatType>(eTy))
+    return failure();
+
+  Location loc = op->getLoc();
+  auto constOneFloat = createFloatConst(loc, operandTy, 1.0, rewriter);
+  auto sqrtOp = rewriter.create<math::SqrtOp>(loc, op->getOperand(0));
+  auto fdivOp = rewriter.create<arith::DivFOp>(
+      loc, operandTy, ValueRange{constOneFloat, sqrtOp});
+  rewriter.replaceOp(op, fdivOp);
```

Comment:
nit: this can be simplified to
```suggestion
  rewriter.replaceOpWithNewOp<arith::DivFOp>(
      op, operandTy, ValueRange{constOneFloat, sqrtOp});
```